### PR TITLE
Update router to hyperapp's latest.

### PR DESCRIPTION
**Pull request by jbucaran**
- No breaking changes.
- Add new .index property to route info object.

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -1,70 +1,84 @@
...
-      match: match,
-      params: params
-    }
+    var route = routes[i][0]
+    var keys = []
+
+    data.replace(
+      RegExp(
+        route === "*"
+          ? "." + route
```
Comment (lukejacksonn) [inline src/router.js]: Someone did say, how come this isn't hard coded to `? ".*"`

Comment (jbucaran) [inline src/router.js]: Saves 2 bytes, but it's okay, let's ".*".

File: src/router.js [hunk 2]
```diff
@@ -1,70 +1,84 @@
...
-        match: function(state, actions, data, emit) {
+        update: function(state, actions, router) {
           return {
-            router: emit("route", match(data))
+            router: router
           }
         },
-        go: function(state, actions, data) {
-          history.pushState({}, "", data)
-          actions.router.match(data.split("?")[0])
```
Comment (lukejacksonn) [inline src/router.js]: Is this case covered in the refactor?

Comment (jbucaran) [inline src/router.js]: @lukejacksonn Nope. We need to decide what the behavior is going to be.

Comment (lukejacksonn) [inline src/router.js]: So urls with query strings should certainly not break the router.. I guess we have two options:

1. Split the url at `?` take the first item in the split array and extract matched parts as `router.params` object. Ignore the second item in split array.
2. Split the url at `?` take the first item in the split array and extract matched parts as `router.params` object **and** take the second item in split array and extract parts as `router.query` object.

**Pull request merged**